src/main/java/org/spongepowered/api/text/Texts.java

```diff
@@ -22,106 +22,170 @@
      */
-    public static MessageBuilder.Translatable builder(Translation translation, Object... args) {
+    public static TextBuilder.Translatable builder(Translation translation, Object... args) {
         return factory.createTranslatableBuilder(translation, args);
     }
 
     /**
-     * Creates a {@link MessageBuilder.Translatable} with the specified
+     * Creates a {@link TextBuilder.Translatable} with the specified
      * {@link Translatable} object and arguments.
      *
      * @param translatable The translatable object to insert to the message
      * @param args The arguments for the translation, can be empty
      * @return A new message builder with the translation of the translatable
      *         object
-     * @see Message.Translatable
+     * @see Text.Translatable
      */
-    public static MessageBuilder.Translatable builder(Translatable translatable, Object... args) {
+    public static TextBuilder.Translatable builder(Translatable translatable, Object... args) {
         return builder(translatable.getTranslation(), args);
     }
 
     // TODO: Change to builder() when possible?
 
     /**
-     * Creates a new {@link MessageBuilder.Selector} with the specified
+     * Creates a new {@link TextBuilder.Selector} with the specified
      * selector.
      *
      * @param selector The selector for the message
      * @return A new message builder with the specified selector
-     * @see Message.Selector
+     * @see Text.Selector
      */
-    public static MessageBuilder.Selector selector(String selector) {
+    public static TextBuilder.Selector selector(String selector) {
         return factory.createSelectorBuilder(selector);
     }
 
     /**
-     * Creates a new {@link MessageBuilder.Score} with the specified score.
+     * Creates a new {@link TextBuilder.Score} with the specified score.
      *
      * @param score The score for the message
      * @return A new message builder with the specified score
-     * @see Message.Score
+     * @see Text.Score
      */
-    public static MessageBuilder.Score score(Object score) {
+    public static TextBuilder.Score score(Object score) {
         return factory.createScoreBuilder(score);
     }
 
     /**
-     * Creates a {@link Message} with the specified plain text. The created
+     * Creates a {@link Text} with the specified plain text. The created
      * message won't have any formatting or events configured.
      *
      * @param content The content of the Message
-     * @return The created {@link Message}
+     * @return The created {@link Text}
      */
-    public static Message.Text of(String content) {
+    public static Text.Literal of(String content) {
         return factory.createPlain(content);
     }
 
     /**
+     * Joins a sequence of text objects together.
+     *
+     * @param texts The texts object
+     * @return A text object that joins the given text objects
+     */
+    public static Text join(Text... texts) {
```

Comment:
Maybe also Join with a seperator? Similar to the one in I believe Apache commons stringutils class